### DISCUSSION TITLE
<compare> not found during the build.

### DISCUSSION THREAD

**Discussion by YuanSha0** [AliveToolkit/alive2#935] [Q&A]
My GCC version is 9.4.0, and I initially suspected that the GCC version was too low. Later, I changed the compiler to the latest version of Clang, but the error still persists. Can anyone help me? I will provide the terminal output next.

alive2/smt/expr.h:6:10: fatal error: compare: No such file or directory
    6 | #include <compare>
      |          ^~~~~~~~~

**Comment by nunoplopes** [comment]
You need gcc 10 at least.
With clang, you may need to pass `-stdlib=libc++` to CXXFLAGS to prevent clang from using gcc's headers.
